### PR TITLE
Fix outdated comments and documentation, and split up a file.

### PR DESCRIPTION
**Pull request by gunney1**
# Documentation fix for `DistributedClosestPoint`

- This PR is a documenation fix and a move of an implementation class into its own file.
- It does the following (modify list as needed):
  - Fix outdated comments
  - Move the implementation class `DistributedClosestPointImpl` into file `detail/DistributedClosestPointImpl.hpp`.
  - Document a slight interface change in `RELEASE-NOTES.md`.

I split up the `DistributedClosestPoint.hpp` file because it was too big and confusingly had some methods with similar names but in different classes.

### CODE REVIEW

File: src/axom/quest/DistributedClosestPoint.hpp [hunk 1]
```diff
@@ -1695,12 +328,29 @@ class DistributedClosestPoint
    * \brief Computes the closest point on the object mesh for each point
    * on the provided query mesh
    *
-   * \param [in] query_node conduit node containing the query points
-   * \param [in] coordset The name of the coordinates within query_node
+   * \param [in] query_node The root node of a mesh blueprint for the query points
+   * Can be empty if there are no query points for the calling rank
```
Comment (rhornung67) [inline src/axom/quest/DistributedClosestPoint.hpp:332]:
```suggestion
   * can be empty if there are no query points for the calling rank.
```

File: src/axom/quest/DistributedClosestPoint.hpp [hunk 2]
```diff
@@ -1695,12 +328,29 @@ class DistributedClosestPoint
...
+   * \param [in] topology The name of the topology within query_node
+   *
+   * @c queryMesh should have data on the host, regardless of the runtime
+   * policy setting.  Data will be copied to device as needed.
+   *
+   * On completion, the query mesh contains the following fields:
+   *   - cp_rank: Will hold the rank of the object point containing the closest point
+   *   - cp_domain_index: will hold the index of the object domain containing
```
Comment (rhornung67) [inline src/axom/quest/DistributedClosestPoint.hpp:340]:
```suggestion
   *   - cp_domain_index: Will hold the index of the object domain containing
```

Consistency with other lines.

File: src/axom/quest/DistributedClosestPoint.hpp [hunk 3]
```diff
@@ -1695,12 +328,29 @@ class DistributedClosestPoint
...
+   *     For multiple object mesh domains on a rank, cp_index is relative to
+   *     each domain.
+   *   - cp_coords: Will hold the coordinates of the closest points
+   *     interleaved in a 1D array.
+   *   - cp_distance: Will hold the distances to the closest points.
+   * See setOutput() to toggle these outputs.
    *
-   * \pre query_node must follow the conduit mesh blueprint convention
+   * \note The current implementation assumes that the mesh coordinates
+   * interleaved or contiguous.  The output cp_coords will be contiguous.
```
Comment (rhornung67) [inline src/axom/quest/DistributedClosestPoint.hpp:351]:
```suggestion
   * are interleaved or contiguous.  The output cp_coords will be contiguous.
```

File: src/axom/quest/detail/DistributedClosestPointImpl.hpp [hunk 4]
```diff
@@ -0,0 +1,1350 @@
...
+#ifndef AXOM_USE_MPI
+  #error This file requires Axom to be configured with MPI
+#endif
+#include "mpi.h"
+
+namespace axom
+{
+namespace quest
+{
+namespace internal
```
Comment (rhornung67) [inline src/axom/quest/detail/DistributedClosestPointImpl.hpp:39]: I'm confused. You've moved all of these methods from one header file to another. Most of these will not be inlined and, if included, in multiple header files may be parsed multiple times. Would it be better to move these to a source (*.cpp) file?

Comment (gunney1) [reply src/axom/quest/detail/DistributedClosestPointImpl.hpp:39]: @rhornung67 That's a good point with respect to the `DistributedClosestPoint` class.  Its methods can be moved a `DistributedClosestPoint.cpp` file.  The templated `DistributedClosestPointImpl` class will then be parsed only once, when included in the `.cpp` file.  Let me know if that doesn't address your point.

File: src/axom/quest/DistributedClosestPoint.hpp [hunk 5]
```diff
@@ -7,25 +7,12 @@
 #define QUEST_DISTRIBUTED_CLOSEST_POINT_H_
 
 #include "axom/config.hpp"
-#include "axom/core.hpp"
-#include "axom/slic.hpp"
```
Comment (gunney1) [inline src/axom/quest/DistributedClosestPoint.hpp:11]: Most of the changes in this file is just moving code into the .cpp file.

**Review by kennyweiss — APPROVED** [review]

**Review by rhornung67 — APPROVED** [review]

**Review by agcapps — APPROVED** [review]

**Pull request merged**